impl.go

```diff
@@ -287,16 +295,23 @@ func validReceiver(recv string) bool {
 }
 
 func main() {
-	if len(os.Args) != 3 {
+	flag.Parse()
+
+	if len(flag.Args()) < 2 {
 		fmt.Fprint(os.Stderr, usage)
 		os.Exit(2)
 	}
-	recv, iface := os.Args[1], os.Args[2]
+
+	recv, iface := flag.Args()[0], flag.Args()[1]
 	if !validReceiver(recv) {
 		fatal(fmt.Sprintf("invalid receiver: %q", recv))
 	}
 
-	fns, err := funcs(iface)
+	if *sourceDirectory == "" {
+		*sourceDirectory, _ = os.Getwd()
```

Comment:
If err is non-nil, we should ignore the other return value, unless that behavior is specifically documented:

```go
if dir, err := os.Getwd(); err != nil {
  *sourceDirectory = dir
}
```